### ISSUE TITLE
Loading NEI saved inventory after adding Electricraft crashes client.

### ISSUE THREAD

**Issue by lossycrypt** [ReikaKalseki/Reika_Mods_Issues#313]
Unexpected client crash on my testing client. What i did:
1) Start client before adding Electricraft.
2) Create new map, cheat in some stuff.
3) Use NEI to save inventory. (lefthand save button)
4) Close client. Add Electricraft. Start again.
5) Create a new world. Press "Load" on NEI.
= Crash

```
`[15:00:15] [Client thread/FATAL]: Reported exception thrown!
net.minecraft.util.ReportedException: Rendering item
    at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:604) ~[bny.class:?]
    at net.minecraft.client.gui.GuiIngame.func_73832_a(GuiIngame.java:955) ~[bbv.class:?]
    at net.minecraftforge.client.GuiIngameForge.renderHotbar(GuiIngameForge.java:206) ~[GuiIngameForge.class:?]
    at net.minecraftforge.client.GuiIngameForge.func_73830_a(GuiIngameForge.java:141) ~[GuiIngameForge.class:?]
    at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1038) ~[blt.class:?]
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001) ~[bao.class:?]
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898) [bao.class:?]
    at net.minecraft.client.main.Main.main(SourceFile:148) [Main.class:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.7.0_80]
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_80]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_80]
    at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.7.0_80]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.11.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.11.jar:?]
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
    at java.util.ArrayList.rangeCheck(Unknown Source) ~[?:1.7.0_80]
    at java.util.ArrayList.get(Unknown Source) ~[?:1.7.0_80]
    at java.util.Collections$UnmodifiableList.get(Unknown Source) ~[?:1.7.0_80]
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getExtractType(ItemCustomModOre.java:99) ~[ItemCustomModOre.class:?]
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getOreType(ItemCustomModOre.java:94) ~[ItemCustomModOre.class:?]
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getColorOne(ItemCustomModOre.java:60) ~[ItemCustomModOre.class:?]
    at Reika.DragonAPI.Auxiliary.ReikaSpriteSheets.renderItem(ReikaSpriteSheets.java:114) ~[ReikaSpriteSheets.class:?]
    at Reika.DragonAPI.Instantiable.Rendering.ItemSpriteSheetRenderer.renderItem(ItemSpriteSheetRenderer.java:55) ~[ItemSpriteSheetRenderer.class:?]
    at net.minecraftforge.client.ForgeHooksClient.renderInventoryItem(ForgeHooksClient.java:183) ~[ForgeHooksClient.class:?]
    at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:563) ~[bny.class:?]
    ... 13 more
[15:00:15] [Client thread/INFO] [STDOUT]: [net.minecraft.client.Minecraft:func_71377_b:349]: ---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 15/10/06 15:00
Description: Rendering item

java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
    at java.util.ArrayList.rangeCheck(Unknown Source)
    at java.util.ArrayList.get(Unknown Source)
    at java.util.Collections$UnmodifiableList.get(Unknown Source)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getExtractType(ItemCustomModOre.java:99)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getOreType(ItemCustomModOre.java:94)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getColorOne(ItemCustomModOre.java:60)
    at Reika.DragonAPI.Auxiliary.ReikaSpriteSheets.renderItem(ReikaSpriteSheets.java:114)
    at Reika.DragonAPI.Instantiable.Rendering.ItemSpriteSheetRenderer.renderItem(ItemSpriteSheetRenderer.java:55)
    at net.minecraftforge.client.ForgeHooksClient.renderInventoryItem(ForgeHooksClient.java:183)
    at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:563)
    at net.minecraft.client.gui.GuiIngame.func_73832_a(GuiIngame.java:955)
    at net.minecraftforge.client.GuiIngameForge.renderHotbar(GuiIngameForge.java:206)
    at net.minecraftforge.client.GuiIngameForge.func_73830_a(GuiIngameForge.java:141)
    at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1038)
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
    at java.util.ArrayList.rangeCheck(Unknown Source)
    at java.util.ArrayList.get(Unknown Source)
    at java.util.Collections$UnmodifiableList.get(Unknown Source)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getExtractType(ItemCustomModOre.java:99)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getOreType(ItemCustomModOre.java:94)
    at Reika.RotaryCraft.ModInterface.ItemCustomModOre.getColorOne(ItemCustomModOre.java:60)
    at Reika.DragonAPI.Auxiliary.ReikaSpriteSheets.renderItem(ReikaSpriteSheets.java:114)
    at Reika.DragonAPI.Instantiable.Rendering.ItemSpriteSheetRenderer.renderItem(ItemSpriteSheetRenderer.java:55)
    at net.minecraftforge.client.ForgeHooksClient.renderInventoryItem(ForgeHooksClient.java:183)

-- Item being rendered --
Details:
    Item Type: Reika.RotaryCraft.ModInterface.ItemCustomModOre@6c287457
    Item Aux: 2
    Item NBT: {bedrock:1b,creative:1b,energy:4800000000000000L}
    Item Foil: false
Stacktrace:
    at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:563)
    at net.minecraft.client.gui.GuiIngame.func_73832_a(GuiIngame.java:955)
    at net.minecraftforge.client.GuiIngameForge.renderHotbar(GuiIngameForge.java:206)
    at net.minecraftforge.client.GuiIngameForge.func_73830_a(GuiIngameForge.java:141)

-- Affected level --
Details:
    Level name: MpServer
    All players: 1 total; [EntityClientPlayerMP['Farbstoff'/48, l='MpServer', x=-1845.50, y=5.62, z=463.50]]
    Chunk stats: MultiplayerChunkCache: 441, 441
    Level seed: 0
    Level generator: ID 01 - flat, ver 0. Features enabled: false
    Level generator options: 
    Level spawn location: World: (-1836,4,455), Chunk: (at 4,0,7 in -115,28; contains blocks -1840,0,448 to -1825,255,463), Region: (-4,0; contains chunks -128,0 to -97,31, blocks -2048,0,0 to -1537,255,511)
    Level time: 227 game time, 227 day time
    Level dimension: 0
    Level storage version: 0x00000 - Unknown?
    Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
    Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
    Forced entities: 1 total; [EntityClientPlayerMP['Farbstoff'/48, l='MpServer', x=-1845.50, y=5.62, z=463.50]]
    Retry entities: 0 total; []
    Server brand: fml,forge
    Server type: Integrated singleplayer server
Stacktrace:
    at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:373)
    at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2444)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:919)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
    Minecraft Version: 1.7.10
    Operating System: Windows 8 (amd64) version 6.2
    Java Version: 1.7.0_80, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 723630880 bytes (690 MB) / 1541406720 bytes (1470 MB) up to 1908932608 bytes (1820 MB)
    JVM Flags: 1 total; -Xmx2G
    AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
    IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
    FML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1448 16 mods loaded, 16 mods active
    States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
    UCHIJAAAA   mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
    UCHIJAAAA   FML{7.10.99.99} [Forge Mod Loader] (forge-1.7.10-10.13.4.1448-1.7.10.jar) 
    UCHIJAAAA   Forge{10.13.4.1448} [Minecraft Forge] (forge-1.7.10-10.13.4.1448-1.7.10.jar) 
    UCHIJAAAA   CodeChickenCore{1.0.7.47} [CodeChicken Core] (minecraft.jar) 
    UCHIJAAAA   NotEnoughItems{1.0.5.111} [Not Enough Items] (NotEnoughItems-1.7.10-1.0.5.111-universal.jar) 
    UCHIJAAAA   <CoFH ASM>{000} [CoFH ASM] (minecraft.jar) 
    UCHIJAAAA   <DragonAPI ASM>{0} [DragonAPI ASM Data Initialization] (minecraft.jar) 
    UCHIJAAAA   CoFHCore{1.7.10R3.0.3} [CoFH Core] (CoFHCore-[1.7.10]3.0.3-303.jar) 
    UCHIJAAAA   DragonAPI{1.0} [DragonAPI] (DragonAPI 1.7.10 V9b.jar) 
    UCHIJAAAA   ChromatiCraft{1.0} [ChromatiCraft] (ChromatiCraft 1.7.10 V9b.jar) 
    UCHIJAAAA   RotaryCraft{1.0} [RotaryCraft] (RotaryCraft 1.7.10 V9b.jar) 
    UCHIJAAAA   ElectriCraft{1.0} [ElectriCraft] (ElectriCraft 1.7.10 V9b.jar) 
    UCHIJAAAA   ImmersiveEngineering{0.5.2} [Immersive Engineering] (ImmersiveEngineering-0.5.2.jar) 
    UCHIJAAAA   journeymap{@JMVERSION@} [JourneyMap] (journeymap-1.7.10-5.1.0-unlimited.jar) 
    UCHIJAAAA   ReactorCraft{1.0} [ReactorCraft] (ReactorCraft 1.7.10 V9b.jar) 
    UCHIJAAAA   rftools{3.31} [RFTools] (rftools-3.31.jar) 
    GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.5.0 NVIDIA 355.60' Renderer: 'GeForce GT 650M/PCIe/SSE2'
    CoFHCore: -[1.7.10]3.0.3-303
    Launched Version: MultiMC5
    LWJGL: 2.9.1
    OpenGL: GeForce GT 650M/PCIe/SSE2 GL version 4.5.0 NVIDIA 355.60, NVIDIA Corporation
    GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

    Is Modded: Definitely; Client brand changed to 'fml,forge'
    Type: Client (map_client.txt)
    Resource Packs: [Soartex_1710_Fanver_Modded_Universal_fixed.zip, BDcraft Sounds Pack.zip]
    Current Language: English (US)
    Profiler Position: N/A (disabled)
    Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
    Anisotropic Filtering: On (2)
[15:00:15] [Client thread/INFO] [STDOUT]: [net.minecraft.client.Minecraft:func_71377_b:359]: #@!@# Game crashed! Crash report saved to: #@!@# E:\minecraft\nolauncher-bergwerk2.0\instances\bergwerk2\minecraft\crash-reports\crash-2015-10-06_15.00.15-client.txt
[15:00:15] [Client Shutdown Thread/INFO]: Stopping server
Exception in thread "ChromatiCraft Structure Gen" AL lib: (EE) alc_cleanup: 1 device not closed

```

**Comment by ReikaKalseki** [comment]
ItemCustomModOre is for those ores added in the special config file. Those items, if hacked in, will cause a crash no matter what.

Unless you spawned none of them in, in which case adding a mod scrambled your IDs.

**Comment by Reignofmagic** [comment]
NEI Inventory save/load data is a little funky. Adding new mods, removing mods, or even starting a new world will change the items given. Sometimes drastically, additionally this might cause a crash. Just delete your saved inventory list (in nei) and make new ones after you change your modlist. (A way to save them is to put the inventory saves in different chests prior to changing modlists.)

For example one time I had an inventory of a super builder wand, creative energy cell, and a few other random gizmos I use in test worlds. When I made a new world, none of the items from the previous world were there when I used the NEI inventory load function. I got other completely random items. This is because the Item ID's are saved and generated on a per world basis instead of being set in stone like they were in 1.6.4
